### DISCUSSION TITLE
Update dynamically the token for Facebook ads connection

### DISCUSSION THREAD

**Discussion by mkhalil7** [apache/airflow#18769] [Q&A]
Hello, 

I am currently building a pipeline that extract daily the ads data for Facebook using airflow.providers.facebook.ads.hooks.ads

Currently the connection parameter including the Facebook Token are hard-coded into the connection parameter of Airflow

The problem is we need to change regularly the token into the connection parameters of airflow. 

We can receive the new Token using a rest call

Is there a way for us to modify from the DAG code the connection parameter or to pass dynamically the token to the Facebookadshook ?

**Comment by potiuk** [comment]
You could use secret backend to store your connections: https://airflow.apache.org/docs/apache-airflow/stable/security/secrets/secrets-backend/index.html - there are couple of existing community provided security backends (GCP/Azure/AWS/Hashicorp) and you can roll your own. Then you can update the connection with new token directly in the secret backend when you make your REST call, and next time the connection will be retrieved using the new token.
